File: Pessoa.rb
class Pessoa

  def initialize(idade, peso, altura)
    @idade = idade
    @peso = peso
    @altura = altura
  end

  def engordar(quantidade)
    @peso += quantidade
  end

  def emagrecer(quantidade)
    @peso -= quantidade
  end

  def envelhecer()
    if @idade < 21
      @idade += 1
      crescer()
    elsif
      @idade += 1
      obter_altura()
    end
  end

  def obter_idade()
    @idade
  end

  def obter_peso()
    @peso
  end

  def obter_altura()
    @altura
  end

private

  def crescer()
    @altura += 0.015
  end
end

File: Pessoa.java

public class Pessoa {

	private double peso;
	private int idade;
	private double altura;

	public Pessoa(double peso, int idade, double altura) {
		this.peso = peso;
		this.idade = idade;
		this.altura = altura;
	}

	public double peso() {
		return this.peso;
	}

	public int idade() {
		return this.idade;
	}

	public double altura() {
		return this.altura;
	}

	public void engordar(double pesoEngordado) {
		this.peso += pesoEngordado;		
	}

	public void emagrecer(double pesoEmagrecido) {
		this.peso -= pesoEmagrecido;		
	}

	public void envelhecer() {
		this.idade += 1;
		crescer();
	}
	
	private void crescer(){
		if (this.idade <= 21)
			this.altura += 0.015;
	}
}
